Source/ASDisplayNode.mm

```diff
@@ -572,10 +572,11 @@ - (void)_didLoad
   for (ASDisplayNodeDidLoadBlock block in onDidLoadBlocks) {
     block(self);
   }
-
-  for (id <ASInterfaceStateDelegate> delegate in _interfaceStateDelegates) {
-    if ([delegate respondsToSelector:@selector(nodeDidLoad)]) {
-      [delegate nodeDidLoad];
+  @synchronized (_interfaceStateDelegates) {
```

Comment:
Please use `__instanceLock__`.

In some methods, like this one, the lock is acquired earlier (i.e a few lines above). You can take advantage of that by storing the delegates in a local variable while the lock is being held, then use it a bit later on.

Reply:
delegates calling like didEnterVisibleState is requring ASAssertUnlocked(__instanceLock__);

Reply:
Ah yeah, that just means that method requires to be called without the lock, but within the method it can acquire the lock itself.

Reply:
__instanceLock__.lock();
   NSHashTable  * **delegates** = [_interfaceStateDelegates copy];
   __instanceLock__.unlock();
...
ASAssertUnlocked(__instanceLock__);
...
for (id <ASInterfaceStateDelegate> delegate in **delegates**) {
      if ([delegate respondsToSelector:@selector(nodeDidLoad)]) {
        [delegate nodeDidLoad];
      }
    }
Will something like this work

Reply:
Yes.